File: packages/core/frontend/src/types/resources/endpointslice.ts
import type { ManagedField } from "./shared";

export interface EndpointSliceEndpoint {
  Addresses: string[];
  Hostname: string;
  NodeName: string;
  Zone: string;
  TargetName: string;
  TargetKind: string;
  Ready: boolean;
  Serving: boolean;
  Terminating: boolean;
}

export interface EndpointSlicePort {
  Name: string;
  Port: number;
  Protocol: string;
}

export interface EndpointSlice {
  Name: string;
  Namespace: string;
  AddressType: string;
  Age: string;
  Ports: EndpointSlicePort[];
  Endpoints: EndpointSliceEndpoint[];
  // detail fields
  CreatedAt: string;
  ControlledBy: string;
  ServiceName: string;
  Labels: Record<string, string>;
  Annotations: Record<string, string>;
  ManagedFields: ManagedField[];
}

File: packages/core/kube/dto/endpointslice.go
package dto

type EndpointSliceRef struct {
	Namespace string `json:"namespace"`
	Name      string `json:"name"`
}

type EndpointSliceEndpoint struct {
	Addresses   []string
	Hostname    string
	NodeName    string
	Zone        string
	TargetName  string
	TargetKind  string
	Ready       bool
	Serving     bool
	Terminating bool
}

type EndpointSlicePort struct {
	Name     string
	Port     int32
	Protocol string
}

type EndpointSlice struct {
	// list view
	Name        string
	Namespace   string
	AddressType string
	Ports       []EndpointSlicePort
	Endpoints   []EndpointSliceEndpoint
	Age         string
	// detail view
	CreatedAt     string
	Labels        map[string]string
	Annotations   map[string]string
	ManagedFields []ManagedField
	ControlledBy  string
	ServiceName   string
}
